internal/command/deploy/deploy.go

```diff
@@ -166,9 +166,16 @@ func DeployWithConfig(ctx context.Context, appConfig *appconfig.Config, args Dep
 		if flag.GetString(ctx, flag.RegionName) != "" {
 			primaryRegion = flag.GetString(ctx, flag.RegionName)
 		}
+
+		processConfigs, err := appConfig.GetProcessConfigs()
+		if err != nil {
+			return err
+		}
+
 		md, err := NewMachineDeployment(ctx, MachineDeploymentArgs{
 			AppCompact:        appCompact,
 			DeploymentImage:   img,
+			ProcessConfigs:    &processConfigs,
```

Comment:
I'm curious why did you prefer to pass `processConfigs` from the "outside" instead of grabbing them from the context's appConfig

Reply:
no particular reason! that makes a lot more sense though, and looking at it with fresh eyes I think the current method would (maybe?) break when `secrets` deploys because it doesn't set that. I'll switch it up tomorrow morning :)